CRM/Upgrade/Incremental/sql/5.4.alpha1.mysql.tpl

```diff
@@ -11,3 +11,11 @@ we want to ensure that these values are handled precisely and consistently.
 ALTER TABLE civicrm_cache
   CHANGE created_date created_date TIMESTAMP DEFAULT CURRENT_TIMESTAMP COMMENT 'When was the cache item created',
   CHANGE expired_date expired_date  TIMESTAMP NULL DEFAULT NULL COMMENT 'When should the cache item expire';
+
+
+-- core/issues/230
+ALTER TABLE `civicrm_saved_search`
+  DROP FOREIGN KEY FK_civicrm_saved_search_mapping_id;
```

Comment:
@eileenmcnaughton is there a php function we have for doing this sort of work?

Reply:
we have add/remove column function in php

Reply:
@sunilpawar @seamuslee001 it should be possible to get the foreign key fixed by running the api
System.updateindexes

As a general rule we stopped updating indexes in upgrade scripts with the intention that sites can update them in their own time using System.updateindexes (e.g via drush).

Originally we had a System.Check to encourage this. Unfortunately some older sites had weird & wonderful indexes we couldn't handle. We were working on handling those & I think @aydun did a good analysis but after @agh1 suggested we drop the System.check due to the sites it didn't work on the momentum died.

However, I still think the principle is correct - we should not be altering indexes any more in the upgrade script but getting people to run that script. We probably need to either re-enable the check with improved error handling & soft wording about how this is a nice to fix or add a message to the upgrade script. However, we don't have to solve all that here

==tldr - I'm happy to merge the xml change (which will affect new sites or those that run the script) but let's remove the upgrade portion of it.

Reply:
If we still did any alterations given this an FK i would strongly recommend using https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/BAO/SchemaHandler.php#L681 and maybe then running the create.

Reply:
Hopefully that is called from the update index - I haven't looked in a while but I think it is

Reply:
I suspect not because this is a foreign key as well as an index and iirc that was one of the problems with the update indexes was it was trying to manipulate indexes that were used as part of FKs as well

Reply:
@seamuslee001 right - but 
a) that is the thing that needs fixing with that approach & hence we should fix
b) I thought it was only WRT dropping some old weirdo indexes from odd configs (ie. create might be OK)

Reply:
@eileenmcnaughton , i don't have issue with removing upgrade part but one question: i don't see any thing in
CRM/Core/BAO/SchemaHandler.php which handle 'on delete set null' action

Reply:
@eileenmcnaughton should i delete the upgrade part (even schema handler is not present for 'on delete set null')?